File: BruteForce/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow)
{
    vector<int> answer;

    for (int i = yellow; i >= 1; i--)
    {
        if (yellow % i == 0)
        {
            if ((i + 2) * ((yellow / i) + 2) == brown + yellow)
            {
                answer.push_back(i + 2);
                answer.push_back((yellow / i) + 2);
                break;
            }
        }
    }

    return answer;
}

File: BruteForce/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int sum = brown + yellow;
        
        for (int i = 3; i < sum; i++) {
            int j = sum/i;
            
            if (sum % i == 0 && j >= 3) {
                int max = Math.max(i, j);
                int min = Math.min(i, j);
                int mid = (min - 2) * (max - 2);
                if (mid == yellow) {
                    answer[0] = max;
                    answer[1] = row;
                    return answer;
                }
            }
        }
        return answer;
    }
}s